pkg/xray/provider_test.go

```diff
@@ -41,17 +43,50 @@ func getTestResty(t *testing.T) *resty.Client {
 	return restyClient
 }
 
-func testAccPreCheck(t *testing.T) {
+func testAccPreCheck(t *testing.T) error {
 	restyClient := getTestResty(t)
-	// TODO check the payload and make sure it's the right license type
-	_, err := restyClient.R().Get("/artifactory/api/system/licenses/")
+	err := checkArtifactoryLicense(restyClient)
 	if err != nil {
-		t.Fatal(err)
+		return err
 	}
 	ctx := context.Background()
 	provider, _ := testAccProviders["xray"]()
 	oldErr := provider.Configure(ctx, terraform.NewResourceConfigRaw(nil))
 	if oldErr != nil {
 		t.Fatal(oldErr)
 	}
+	return nil
+}
+
+// Create a local repository with Xray indexing enabled. It will be used in the tests
+func testAccCreateRepos(t *testing.T, repos []string) {
+	restyClient := getTestResty(t)
+	body := "{\n\"rclass\":\"local\",\n\"xrayIndex\":true\n}"
+	for _, repo := range repos {
+		_, errRepo := restyClient.R().SetBody(body).Put("artifactory/api/repositories/" + repo)
+		repoExists := strings.Contains(fmt.Sprint(errRepo), "Case insensitive repository key already exists")
+		repoCreated := strings.Contains(fmt.Sprint(errRepo), "Successfully created repository")
```

Comment:
for repoExists call the response code is 400, which could happen for some other reason as well. This is why I'm checking the body to make sure that 400 was triggered by the fact, that the repo already exist. 
For repoCreated I can check the response status. Will make this change

Reply:
The output messaging could change at anytime, but the status code won't. So, 400/404 are known RT codes for this

Reply:
Example: 
`            "status": 400,
            "message": "No valid type of repository found.\n"`
and 
`        {
            "status": 400,
            "message": "error when validating repository name: Case insensitive repository key already exists"
        }`
I need to know that the repo already exists, so the test can continue. Otherwise, if I get 400 because the body is incorrect, the test (which requires the repo to exist) will fail.